airport-challenge-solution.md

```diff
@@ -0,0 +1,158 @@
+
+## Tests 1-3
+
+a Need an Airport
+b Need to be able to give the Airport a Plane
+c LandedPlanes list length should increase by 1 when a Plane is added to the empty LandedPlanes list
+
+**Test 1** - `landedPlanes` length increases by 1 when `landPlane` is called with _instance of `Plane`_
+
+**Test 2** - `landPlane` should only add `Plane` instances to the `landedPlanes`
+
+**Test 3** - edge Case - falsy values should not be added to the Airport
+
+//I am not clear if if Test 2 and 3 are the same explicitly?
+
+## Part 2 User Story
+
+As the system designer
+So that the software can be used for many different airports
+I would like a default airport capacity that can be overridden as appropriate
+
+## Part 2 Domain Model (added to from above)
+
+| Objects | Properties                  | Messages                    | Outputs  |
+| ------- | --------------------------- | --------------------------- | -------- |
+| Airport | landedPlanes @Array[@Plane] | land Plane(@Plane)          | @Void    |
+|         | landedCapacity @Integer     | isAirportFull()             | @Boolean |
+|         |                             | getLandedCapacity()         | @Integer |
+|         |                             | newLandedCapacity(@Integer) | @Void    |
+|         |                             |                             |          |
+| Plane   | id @String                  | getId()                     | @String  |
+
+## Tests 4 & 5
+
+a Need to have an Airport but does not necessarily need to have a Plane in it (getLandedCapacity may be zero)
+//this is not the test. need to make default and make sure it is set up when airport is created
+b use default capacity, run newcapacity and check this changes the default capacity
+
+## Part 3 User Story
+
+As an air traffic controller
+To ensure safety
+I want to prevent landing when the airport is full
+
+## Part 3 Domain Model (not added to) - already fine from above??
+
+| Objects | Properties                  | Messages                    | Outputs  |
+| ------- | --------------------------- | --------------------------- | -------- |
+| Airport | landedPlanes @Array[@Plane] | landPlane(@Plane)           | @Void    |
+|         | landedCapacity @Integer     | isAirportFull()             | @Boolean |
+|         |                             | getLandedCapacity()         | @Integer |
+|         |                             | newLandedCapacity(@Integer) | @Void    |
+|         |                             | getLandedPlaneCount()       | @Integer |
+| Plane   | id @String                  | getId()                     | @String  |
+
+## Tests
+
+a Need to check if is AirportFull True prevents Plane going into 'landedPlanes'
+b Need to check if is AirportFull False allows Plane to go into 'landedPlanes'
+//this approach requires the caller to ask if airport is full first (then do something), whereas would be better to have prevention of landPlane if full
+
```

Comment:
Did not use isAirportFull in the end (Boolean return), as this would have required another method anyway after the return from that.